Example/Example/Purchase/CheckoutHandler.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  CheckoutHandler.swift
+//  Example
+//
+//  Created by Adam Campbell on 16/2/21.
+//  Copyright © 2021 Afterpay. All rights reserved.
+//
+
+import Afterpay
+import Foundation
+
+final class CheckoutHandler: CheckoutV2Handler {
+
+  private let didCommenceCheckoutClosure: () -> Void
+  private let onShippingAddressDidChangeClosure: (ShippingAddress) -> Void
+
+  private var checkoutTokenResultCompletion: CheckoutTokenResultCompletion?
+  private var shippingOptionsCompletion: ShippingOptionsCompletion?
+
+  init(
+    didCommenceCheckout: @escaping () -> Void,
+    onShippingAddressDidChange: @escaping (ShippingAddress) -> Void
+  ) {
+    didCommenceCheckoutClosure = didCommenceCheckout
+    onShippingAddressDidChangeClosure = onShippingAddressDidChange
+  }
+
+  func didCommenceCheckout(completion: @escaping CheckoutTokenResultCompletion) {
+    checkoutTokenResultCompletion = completion
+    didCommenceCheckoutClosure()
+  }
+
+  func provideTokenResult(tokenResult: Result<Token, Error>) {
+    checkoutTokenResultCompletion?(tokenResult)
+    checkoutTokenResultCompletion = nil
+  }
+
+  func shippingAddressDidChange(address: ShippingAddress, completion: @escaping ShippingOptionsCompletion) {
+    shippingOptionsCompletion = completion
+    onShippingAddressDidChangeClosure(address)
+  }
+
+  func provideShippingOptions(shippingOptions: [ShippingOption]) {
+    shippingOptionsCompletion?(shippingOptions)
+    shippingOptionsCompletion = nil
+  }
+
+  func shippingOptionDidChange(shippingOption: ShippingOption) {}
```

Comment:
Why is this function empty??

Reply:
At the moment we don't do anything with the selected shipping option. However a merchant may want to update their own representation with this information